DependencyInjection/Configuration.php

```diff
@@ -137,7 +138,13 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->scalarNode('router')->defaultValue('router')->end()
-                        ->scalarNode('templating')->defaultValue('templating')->end()
+                        ->scalarNode('templating')->defaultValue(function () {
+                            if (Kernel::VERSION_ID >= 40300) {
```

Comment:
The version of the kernel may be different than the version of other installed Symfony components. What component are we actually interested in to make the decision here?

Reply:
Good question; the thought was that if somebody has a kernel >= 4.3 and have not specified `templating` explicitly, they probably--but may not always be--using Twig as this is the configured default. The only other relevant component, I think, is the FrameworkBundle, but is it actually possible to have a different version of the kernel and FrameworkBundle?

There is probably some outlier here where somebody has a strange combination of things in a project for some reason, but we probably can't fully guess at that. In those instances they would just need to manually specify this option.

Reply:
I would not change the default in this PR and keep that for another one. In a followup PR, I would deprecate this option as it's not relevant anymore and should be removed in the next major version of this bundle, where Twig will always be used.

Reply:
Okay; I've reverted that one.

Before I make that next PR, my understanding then is that the default could change along with a deprecation notice in 2.6, and then in yet another PR , remove it entirely for 3.x. (ex: "Specifying a templating engine other than Twig is deprecated in 2.6 and will be removed in 3.0")

Along with that change for 3.x, I assume that we would remove all references to `EngineInterface` which would simplify a lot of the other code. I haven't played around with master for this bundle yet but could certainly do that once I have a Symfony 5 RC running locally.

Reply:
That's correct.